pkg/server/apiextensions.go

```diff
@@ -446,47 +434,46 @@ func decorateCRDWithBinding(in *apiextensionsv1.CustomResourceDefinition, identi
 	return out
 }
 
-// partialMetadataCRD modifies CRD and replaces all version schemas with minimal ones suitable for partial object
+// makePartialMetadataCRD modifies CRD and replaces all version schemas with minimal ones suitable for partial object
 // metadata.
-func partialMetadataCRD(crd *apiextensionsv1.CustomResourceDefinition) {
+func makePartialMetadataCRD(crd *apiextensionsv1.CustomResourceDefinition) {
 	crd.Annotations[annotationKeyPartialMetadata] = ""
 
 	// set minimal schema that prunes everything but ObjectMeta
-	for _, v := range crd.Spec.Versions {
-		v.Schema = &apiextensionsv1.CustomResourceValidation{
+	old := crd.Spec.Versions
+	crd.Spec.Versions = make([]apiextensionsv1.CustomResourceDefinitionVersion, len(old))
```

Comment:
since this is a slice, I think it should be `make([]apiextensionsv1.CustomResourceDefinitionVersion, 0, len(old))`

Reply:
ah, nvm, you index down below